cmake/modules/dts.cmake

```diff
@@ -133,15 +133,27 @@ if(NOT DEFINED DTS_SOURCE)
 endif()
 
 if(EXISTS ${DTS_SOURCE})
-  # We found a devicetree. Check for a board revision overlay.
+  # We found a devicetree. Check for a board revision overlays.
   if(DEFINED BOARD_REVISION)
-    zephyr_build_string(dts_board_string BOARD ${BOARD}
-                                         BOARD_IDENTIFIER ${BOARD_IDENTIFIER}
-                                         BOARD_REVISION ${BOARD_REVISION}
+    zephyr_build_string(
+      dts_board_string
+      BOARD ${BOARD}
+      BOARD_IDENTIFIER ${BOARD_IDENTIFIER}
+      BOARD_REVISION ${BOARD_REVISION}
+      MERGE
```

Comment:
Approach approved, but if using `zephyr_file()` then it should be possible to get rid of the file existence check below as well as simplify the code.
```
zephyr_file(CONF_FILES ${BOARD_DIR} DTS DTS_SOURCE)

zephyr_file(
  CONF_FILES ${BOARD_DIR}
  DTS excluded_dts_board_overlays
  BOARD ${BOARD}
  BOARD_IDENTIFIER ${BOARD_IDENTIFIER}
)

list(REMOVE_ITEM DTS_SOURCE ${excluded_dts_board_overlays})
```

Reply:
Indeed, this is much cleaner ! I have modified things in accordingly.